pygfx/geometries/_base.py

```diff
@@ -107,23 +107,35 @@ def bounding_box(self):
 
         Returns
         -------
-        aabb : ndarray, [2, 3]
+        aabb : ndarray, [2, 3] or None
             The axis-aligned bounding box given by the "smallest" (lowest value)
-            and "largest" (highest value) corners.
+            and "largest" (highest value) corners. Is None when the geometry has
+            no finite positions.
 
         """
         if hasattr(self, "positions"):
             if self._aabb_rev == self.positions.rev:
                 return self._aabb
+            aabb = None
+            # Get positions and check expected shape
             pos = self.positions.data
-            self._aabb = np.array([np.nanmin(pos, axis=0), np.nanmax(pos, axis=0)])
-            # If positions contains xy, but not z, assume z=0
-            if self._aabb.shape[1] == 2:
-                self._aabb = np.column_stack([self._aabb, np.zeros((2, 1), np.float32)])
+            if pos.ndim == 2 and pos.shape[1] in (2, 3):
+                # Select finite positions
+                finite_mask = np.isfinite(pos).all(axis=1)
+                if finite_mask.sum() > 0:
+                    # Construct aabb
+                    pos_finite = pos[finite_mask]
+                    aabb = np.array(
+                        [pos_finite.min(axis=0), pos_finite.max(axis=0)], np.float32
+                    )
+                    # If positions contains xy, but not z, assume z=0
+                    if aabb.shape[1] == 2:
+                        aabb = np.column_stack([aabb, np.zeros((2, 1), np.float32)])
+            self._aabb = aabb
             self._aabb_rev = self.positions.rev
```

Comment:
The change here is to deal with:
* positions being empty.
* positions having inf values.
* positions having no finite values.